cases/filters.py

```diff
@@ -132,12 +132,14 @@ def priority_only_filter(self, queryset, name, value):
 
     def closed_filter(self, queryset, name, value):
         if value == "none":
-            return queryset.filter(closed=False)
+            cs = [c for c in queryset if not c.is_closed_or_merged_into_closed]
```

Comment:
I think this would be the main issue, it loops through every result and checks each one to see if it's closed/merged-into-closed, rather than constructing a query.